Tests/FeedStoreIntegrationTests.swift

```diff
@@ -17,70 +17,63 @@ class FeedStoreIntegrationTests: XCTestCase {
 		super.setUp()
 		
-		setupEmptyStoreState()
+		setupEmptyStoreState(for: self)
 	}
 	
 	override func tearDown() {
 		super.tearDown()
 		
-		undoStoreSideEffects()
+		undoStoreSideEffects(for: self)
 	}
 	
 	func test_retrieve_deliversEmptyOnEmptyCache() {
-		//        let sut = makeSUT()
-		//
-		//        expect(sut, toRetrieve: .empty)
+		let sut = makeSUT()
+	
+		expect(sut, toRetrieve: .empty)
 	}
 	
 	func test_retrieve_deliversFeedInsertedOnAnotherInstance() {
-		//        let storeToInsert = makeSUT()
-		//        let storeToLoad = makeSUT()
-		//        let feed = uniqueImageFeed()
-		//        let timestamp = Date()
-		//
-		//        insert((feed, timestamp), to: storeToInsert)
-		//
-		//        expect(storeToLoad, toRetrieve: .found(feed: feed, timestamp: timestamp))
+		        let storeToInsert = makeSUT()
+		        let storeToLoad = makeSUT()
+		        let feed = uniqueImageFeed()
+		        let timestamp = Date()
+		
+		        insert((feed, timestamp), to: storeToInsert)
+		
+		        expect(storeToLoad, toRetrieve: .found(feed: feed, timestamp: timestamp))
 	}
 	
 	func test_insert_overridesFeedInsertedOnAnotherInstance() {
-		//        let storeToInsert = makeSUT()
-		//        let storeToOverride = makeSUT()
-		//        let storeToLoad = makeSUT()
-		//
-		//        insert((uniqueImageFeed(), Date()), to: storeToInsert)
-		//
-		//        let latestFeed = uniqueImageFeed()
-		//        let latestTimestamp = Date()
-		//        insert((latestFeed, latestTimestamp), to: storeToOverride)
-		//
-		//        expect(storeToLoad, toRetrieve: .found(feed: latestFeed, timestamp: latestTimestamp))
+		let storeToInsert = makeSUT()
+		let storeToOverride = makeSUT()
+		let storeToLoad = makeSUT()
+		
+		insert((uniqueImageFeed(), Date()), to: storeToInsert)
+		
+		let latestFeed = uniqueImageFeed()
+		let latestTimestamp = Date()
+		insert((latestFeed, latestTimestamp), to: storeToOverride)
+		
+		expect(storeToLoad, toRetrieve: .found(feed: latestFeed, timestamp: latestTimestamp))
 	}
 	
 	func test_delete_deletesFeedInsertedOnAnotherInstance() {
-		//        let storeToInsert = makeSUT()
-		//        let storeToDelete = makeSUT()
-		//        let storeToLoad = makeSUT()
-		//
-		//        insert((uniqueImageFeed(), Date()), to: storeToInsert)
-		//
-		//        deleteCache(from: storeToDelete)
-		//
-		//        expect(storeToLoad, toRetrieve: .empty)
+		let storeToInsert = makeSUT()
+		let storeToDelete = makeSUT()
+		let storeToLoad = makeSUT()
+		
+		insert((uniqueImageFeed(), Date()), to: storeToInsert)
+		
+		deleteCache(from: storeToDelete)
+		
+		expect(storeToLoad, toRetrieve: .empty)
 	}
 	
 	// - MARK: Helpers
 	
-	private func makeSUT() -> FeedStore {
-		fatalError("Must be implemented")
-	}
-	
-	private func setupEmptyStoreState() {
-		
-	}
-	
-	private func undoStoreSideEffects() {
-		
+	private func makeSUT(storeUrl: URL? = nil, file: StaticString = #filePath, line: UInt = #line) -> FeedStore {
```

Comment:
The `storeUrl` parameter is never passed to this function. So you can remove it.

```suggestion
	private func makeSUT(file: StaticString = #filePath, line: UInt = #line) -> FeedStore {
```